lib/traject/readers/folio_postgres_reader.rb

```diff
@@ -32,8 +32,15 @@ def each
 
         # declare a cursor
         queries = if @updated_after
-                    conditions = %w[vi hr item cr cl cc].map { |table| "sul_mod_inventory_storage.strtotimestamp((#{table}.jsonb -> 'metadata'::text) ->> 'updatedDate'::text) > '#{@updated_after}'" }.map { |q| [q, @sql_filters].compact }
-                    conditions.map { |c| sql_query(c) }.join(') UNION (')
+                    filter_join = {
+                      'hr_filter' => 'LEFT JOIN sul_mod_inventory_storage.holdings_record hr_filter ON hr_filter.instanceid = vi.id',
+                      'item_filter' => 'LEFT JOIN sul_mod_inventory_storage.item item_filter ON item_filter.holdingsrecordid = hr.id'
+                      'cr_filter'=> 'LEFT JOIN sul_mod_courses.coursereserves_reserves cr_filter ON (cr_filter.jsonb ->> \'itemId\')::uuid = item.id'
+                      'cl_filter'=> 'LEFT JOIN sul_mod_courses.coursereserves_courselistings cl_filter ON cl_filter.id = cr.courselistingid'
+                      'cc_filter'=> 'LEFT JOIN sul_mod_courses.coursereserves_courses cc_filter ON cc_filter.courselistingid = cl.id'
+                    }
+                    conditions = %w[vi hr_filter item_filter cr_filter cl_filter cc_filter].map { |table| "sul_mod_inventory_storage.strtotimestamp((#{table}.jsonb -> 'metadata'::text) ->> 'updatedDate'::text) > '#{@updated_after}'" }.map { |q| [q, @sql_filters].compact }
```

Comment:
Does `HAVING` work (performantly) for the child groupings instead of adding an additional join?

Reply:
I tried a couple variations of HAVING and all of them seemed to be much slower (15s vs minutes). Maybe duplicating the join conditions isn't so bad.